wpilib/wpilib/sendablebuilder.py

```diff
@@ -0,0 +1,234 @@
+        """
+        return self.table
+
+    def updateTable(self):
+        """
+        Update the network table values by calling the getters for all properties.
+        """
+        for prop in self.properties:
+            if prop.update is not None:
+                prop.update(prop.entry)
+
+        if self._updateTable is not None:
+            self._updateTable()
+
+    def startLiveWindowMode(self):
+        """
+        Start LiveWindow mode by hooking the setters for all properties.
+        """
+        if self.safeState is not None:
+            self.safeState()
+
+        for prop in self.properties:
+            prop.startListener()
+
+    def stopLiveWindowMode(self):
+        """
+        Stop LiveWindow mode by unhooking the setters for all properties.
+        """
+        if self.safeState is not None:
+            self.safeState()
+
+        for prop in self.properties:
+            prop.stopListener()
+
+    def setSmartDashboardType(self, type):
+        """
+        Set the string representation of the named data type that will be used
+        by the smart dashboard for this sendable.
+   
+        :param type: data type
+        :type type: str
+        """
+        self.table.getEntry(".type").setString(type)
+
+    def setSafeState(self, func):
+        """
+        Set the function that should be called to set the Sendable into a safe
+        state.  This is called when entering and exiting Live Window mode.
+
+        :param func:    function
+        """
+        self.safeState = func
+
+    def setUpdateTable(self, func):
+        """
+         Set the function that should be called to update the network table
+        for things other than properties.  Note this function is not passed
+        the network table object; instead it should use the entry handles
+        returned by getEntry().
+
+        :param func:    function
+        """
+        self._updateTable = func
+
+    def getEntry(self, key):
+        """
+        Add a property without getters or setters.  This can be used to get
+        entry handles for the function called by setUpdateTable().
+
+        :param key:   property name
+        :type key: str
+        :returns: Network table entry
+        :rtype: :class:`networktables.entry.NetworkTableEntry`
+        """
+        return self.table.getEntry(key)
+
+    def _addProperty(self, key, updater, setter):
+        prop = Property(self.table, key, updater, setter)
+        self.properties.append(prop)
+
+    def addBooleanProperty(self, key, getter, setter):
+        """
+        Add a boolean property.
+   
+        :param key:     property name
+        :type key: str
+        :param getter:  getter function (returns current value)
+        :type getter: typing.Callable[[], bool]
+        :param setter:  setter function (sets new value)
+        :type setter: typing.Callable[[networktables.entry.NetworkTableEntry, str, bool, Any], None]
```

Comment:
need guidance on the right way to express these

Reply:
the right way = not in the docstring

Reply:
I though 3.4 compat prevents it from going elsewhere

Reply:
Is there a compelling reason to support 3.4 still?

Reply:
there's a backport of the typing module in either case

Reply:
Honestly, I would say pick one of the two:

* Use the type annotations
* Build the documentation locally, and change the type to match what you would want it to look like as sphinx renders it. I'm not too picky about it... but it seems like this wouldn't render very well?

Reply:
wow, typing generates some bizarre errors in places I didn't touch. guess we'll have to go with option 2. how do I build the documentation?

Reply:
Sorry, missed your question. Install sphinx and sphinx-rtd-theme, and do `make html`. If you're on Windows, not sure if that will actually work however.